core/src/display_object/movie_clip.rs

```diff
@@ -2485,6 +2493,25 @@ impl<'gc> MovieClip<'gc> {
 
         self.replace_with_movie(context, movie, None);
     }
+
+    pub fn attach_audio(
+        self,
+        context: &mut UpdateContext<'_, 'gc>,
+        netstream: Option<NetStream<'gc>>,
+    ) {
+        let mut write = self.0.write(context.gc_context);
+        if netstream != write.attached_audio {
+            if let Some(old_netstream) = write.attached_audio {
+                old_netstream.was_detached(context);
+            }
+
+            write.attached_audio = netstream;
```

Comment:
I suppose the notification about "was detached" before actually detaching it is not likely to cause any weird unintended side effects... If it's possible at all.

Reply:
We don't fire a NetStream event for this, so the only code that can even observe this order of events is our own.

Reply:
Alright!